rosidl_generator_c/test/test_interfaces.c

```diff
@@ -77,42 +77,68 @@
 
 #define ARRAY_SIZE 7
 
-#define EXPECT_EQ(arg1, arg2) if (arg1 != arg2) exit(1)
-#define EXPECT_NE(arg1, arg2) if (arg1 == arg2) exit(1)
+#define EXPECT_EQ(arg1, arg2) if (arg1 != arg2) return 1
+#define EXPECT_NE(arg1, arg2) if (arg1 == arg2) return 1
 
-void test_primitives(void);
-void test_primitives_default_value(void);
-void test_strings(void);
-void test_primitives_unbounded_arrays(void);
-void test_primitives_bounded_arrays(void);
-void test_primitives_static_arrays(void);
-void test_submessages(void);
+int test_primitives(void);
+int test_primitives_default_value(void);
+int test_strings(void);
+int test_primitives_unbounded_arrays(void);
+int test_primitives_bounded_arrays(void);
+int test_primitives_static_arrays(void);
+int test_submessages(void);
 
 int main(void)
 {
+  int rc = 0;
   fprintf(stderr, "Testing rosidl_generator_c message types...\n");
   fprintf(stderr, "Testing simple primitive message types...\n");
-  test_primitives();
+  if (test_primitives()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing simple primitives with default values...\n");
-  test_primitives_default_value();
+  if (test_primitives_default_value()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing string types...\n");
-  test_strings();
+  if (test_strings()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing primitives unbounded arrays types...\n");
-  test_primitives_unbounded_arrays();
+  if (test_primitives_unbounded_arrays()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing primitives bounded arrays types...\n");
-  test_primitives_bounded_arrays();
+  if (test_primitives_bounded_arrays()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing primitives static arrays types...\n");
-  test_primitives_static_arrays();
+  if (test_primitives_static_arrays()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
   fprintf(stderr, "Testing nested sub-messages...\n");
-  test_submessages();
-  fprintf(stderr, "All tests were good!\n");
-  return 0;
+  if (test_submessages()) {
+    fprintf(stderr, "FAILED\n");
+    rc++;
+  }
+  if (rc != 0) {
+    fprintf(stderr, "Some tests failed!\n");
+  } else {
+    fprintf(stderr, "All tests were good!\n");
```

Comment:
In general we follow the convention warnings and error messages go to `stderr`, everything else to `stdout`.

I only noticed it on this line since it was modified. The other status messages should probably also go to `stdout`.